config/ssoadmin/config.go

```diff
@@ -0,0 +1,36 @@
+package ssoadmin
+
+import (
+	"github.com/crossplane/upjet/pkg/config"
+
+	"github.com/upbound/provider-aws/config/common"
+)
+
+// Configure adds configurations for the ssoadmin group.
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("aws_ssoadmin_account_assignment", func(r *config.Resource) {
+		r.References["principal_id"] = config.Reference{
+			TerraformName:     "aws_identitystore_group",
+			Extractor:         `github.com/crossplane/upjet/pkg/resource.ExtractParamPath("group_id",true)`,
+			RefFieldName:      "PrincipalGroupRef",
+			SelectorFieldName: "PrincipalGroupSelector",
```

Comment:
The reason I put the word group in the field name is to indicate that the selector (unlike the field) will only work if the target is a group. Here's my thinking for why that's appropriate:

* Technically the AccountAssignment `principalId` field can accept the id of either an identitystore Group or an identitystore User.
* Assigning permissions to groups and not users is a near-universal best practice
* Because the identitystore Groups use a provider-generated UUID, it is very difficult to patch in their values without a selector, and a selector provides a very easy experience. In fact, providing the ability to use such a selector in my organization was one of the primary reasons I picked up this piece of work in the first place.
* Users who want to assign a permission set in a particular account to a user and not a group can still do so, just not using the selector, and their experience is no worse than if the selector didn't exist.

Maybe `PrincipalGroupId` would be better? I'm also fine with just leaving them as the default generated names if that's what you'd prefer.

Reply:
The method we follow when defining a ref and selector for a parameter is usually as follows:
- If that parameter points exactly to a certain source, the reference and selector can be defined.
- If it points to more than one resource rather than a specific resource, we generally do not prefer to define a reference and selector. But if there is a common use for it, we can choose to define a reference and selector to the commonly used resource by evaluating user experiences and feedback.

The situation here means that we define a reference and selector according to the second case. In your usage, using `PrincipalGroup` for the reference and selector name is quite logical and understandable because this is only possible when we use the group ID. However, when I think of it as a user who is not familiar with the content and wants to simply create this resource, I think it would be confusing if the name of a parameter is different from the name of the reference and selector. That's why I prefer default names.

If we want to indicate to the user that the reference and selector are defined for the group, maybe we can add a configuration as follows.

```
		r.MetaResource.ArgumentDocs["principal_id"] = "- (Required) An identifier for an object in SSO, such as a user or group(The reference and selector are defined for the group, not for the user)."

```

Reply:
If we continue with the group, we can also give it a name like `PrincipalIdFromGroup` to make it understandable for the user, I leave the choice up to you.

Reply:
I'm not sure if there is a way to edit the document for the reference and selector, we can continue by editing the primary field for now.